doc/Development_Documentation/23_Installation_and_Upgrade/03_System_Setup_and_Hosting/09_Performance_Guide.md

```diff
@@ -1,13 +1,301 @@
+
+Connection Times (ms)
+              min  mean[+/-sd] median   max
+Connect:        0    0   0.4      0       1
+Processing:    56  182  37.1    189     254
+Waiting:       55  182  37.1    189     254
+Total:         56  182  36.9    189     255
+```
+
+### Opcache Preloading
+Opcache preloading was introduced in PHP 7.4, a feature that could improve the performance of your application significantly, by preloading commonly used framework files and keeping the generated byte-code in memory.
+
+In order to enable Opcache preloading, configure these settings in your ini configuration:
+```bash
+opcache.preload=/var/www/home/config/preload.php
+
+; required for opcache.preload:
+opcache.preload_user=www-data
+```
+
+Note: `memory_limit` should be set to at least `200M` when using preloading with Pimcore.
+
+It is also possible to mark which classes should be preloaded or not by using Symfony service tags [container.preload](https://symfony.com/doc/current/reference/dic_tags.html#dic-tags-container-preload) & [container.no_preload](https://symfony.com/doc/current/reference/dic_tags.html#dic-tags-container-nopreload).
+
+#### Benchmarks:
+On running command, `ab -n 100 -c 20 http://localhost/en`
+
+- Before Changes
+```bash
+```
+
+- After Changes
+```bash
+```
+
+### MySQL/MariaDB Optimizations
+Pimcore uses Data Access Object or DAO pattern to isolate the business logic from low level data access operations. These data access operations play a vital role in performance of Pimcore application features requiring read & write to/from the database.
+Therefore, it is important to tune your database configuration to be performant and to exactly fit the needs of your application.
+
+#### Mysql composite index
+If you are using custom database tables for data persistence, then it is beneficial to optimize these tables to have composite indices, so that query optimizer uses the composite index for queries that matches all columns in the index rather than searching the complete table.
+ 
+#### Mysql Buffer Pool
+The buffer pool is an area in main memory where InnoDB caches table and index data as it is accessed. The main purpose of buffer pool is to the performance by improving the response time of data retrieval. The server system variable `innodb_buffer_pool_size` can be set from 70-80% of the total available memory on a dedicated database server with only or primarily InnoDB tables
+
+Please follow this [discussion](https://dba.stackexchange.com/questions/27328/how-large-should-be-mysql-innodb-buffer-pool-size) to decide buffer pool size and update my.cnf:
+```
+[mysqld]
+    innodb_buffer_pool_size=5G
+```
+
+#### Mysql Query Cache
+Mysql server features a Query Cache. When enabled, the query cache stores SELECT statements together with the retrieved record set in memory, then if another identical query is received, the server can then retrieve the results from the query cache rather than parsing and executing the same query again.
```

Comment:
```suggestion
MySQL server features a Query Cache. When enabled, the query cache stores SELECT statements together with the retrieved record set in memory, then if another identical query is received, the server can then retrieve the results from the query cache rather than parsing and executing the same query again.
```